stream/src/main/scala/org/apache/pekko/stream/stage/GraphStage.scala

```diff
@@ -1643,7 +1644,10 @@ abstract class GraphStageLogic private[stream] (val inCount: Int, val outCount:
  *
  * Typical use cases are exchanging messages between stream and substreams or invoking from external world sending
  * event to a stream
+ *
+ * Not for user extension
  */
+@DoNotInherit
```

Comment:
I don't understand this. `def invokeWithFeedback(t: T): Future[Done]` is not implemented so this trait needs to be extended.

Reply:
That's my point. It can be inherited. Can we remove this annotation and maybe change the to say that this is for Pekko Internal Usage?

Reply:
Do you mean add an @InternalApi annotation.

Reply:
Can we just undo the new comment and annotation then?

Reply:
Not sure if applicable but ‘@DoNotInherit` doesn't indicate that it can never be inherit, just that it shouldn't be inherited by users of Pekko. The Pekko library itself can inherit it as much as it wants.

The reason this annotation exists is because of limitations in Java, i.e. Java has no concept of sealed like Scala/Kotlin has

Reply:
The trait has been as is for a while. I am struggling to understand why this PR needs to change the docs and annotations about whether this trait is extensible.

Reply:
@He-Pin what is the reason for not making this user extensible, are you relying on some specific behaviour which could be redefined by a user of Pekko?

Reply:
So I am not seeing why this trait per se would be problematic to extend, it does seem generic enough users could plausibly want to provide their own implementation.

The newly added method however does seem like it shouldn't be extended, in which case you would make it `final`.

Or am I missing something?

Reply:
Why not let users extend and even override this trait? I have no problem with the code changes - just the doc and annotation changes.

Reply:
I will remove that tomorrow but I think this is and only should be called but never be implemented by user